tests/module/utils/test_manim_color.py

```diff
@@ -20,3 +30,136 @@ def test_init_with_int() -> None:
+
+
+def test_to_integer() -> None:
+    color = ManimColor((0x1, 0x2, 0x3, 0x4))
+    nt.assert_equal(color.to_integer(), 0x010203)
+
+
+def test_to_rgb() -> None:
+    color = ManimColor((0x1, 0x2, 0x3, 0x4))
+    nt.assert_array_equal(color.to_rgb(), (0x1 / 255, 0x2 / 255, 0x3 / 255))
+    nt.assert_array_equal(color.to_int_rgb(), (0x1, 0x2, 0x3))
+    nt.assert_array_equal(color.to_rgba(), (0x1 / 255, 0x2 / 255, 0x3 / 255, 0x4 / 255))
+    nt.assert_array_equal(color.to_int_rgba(), (0x1, 0x2, 0x3, 0x4))
+    nt.assert_array_equal(
+        color.to_rgba_with_alpha(0.5), (0x1 / 255, 0x2 / 255, 0x3 / 255, 0.5)
+    )
+    nt.assert_array_equal(
+        color.to_int_rgba_with_alpha(0.5), (0x1, 0x2, 0x3, int(0.5 * 255))
+    )
+
+
+def test_to_hex() -> None:
+    color = ManimColor((0x1, 0x2, 0x3, 0x4))
+    nt.assert_equal(color.to_hex(), "#010203")
+    nt.assert_equal(color.to_hex(True), "#01020304")
+
+
+def test_to_hsv() -> None:
+    color = ManimColor((0x1, 0x2, 0x3, 0x4))
+    nt.assert_array_equal(
+        color.to_hsv(), colorsys.rgb_to_hsv(0x1 / 255, 0x2 / 255, 0x3 / 255)
+    )
+
+
+def test_to_hsl() -> None:
+    color = ManimColor((0x1, 0x2, 0x3, 0x4))
+    nt.assert_array_equal(
+        color.to_hsl(), colorsys.rgb_to_hls(0x1 / 255, 0x2 / 255, 0x3 / 255)
+    )
+
+
+def test_invert() -> None:
+    color = ManimColor((0x1, 0x2, 0x3, 0x4))
+    rgba = color._internal_value
+    inverted = color.invert()
+    nt.assert_array_equal(
+        inverted._internal_value, (1 - rgba[0], 1 - rgba[1], 1 - rgba[2], rgba[3])
+    )
+
+
+def test_invert_with_alpha() -> None:
+    color = ManimColor((0x1, 0x2, 0x3, 0x4))
+    rgba = color._internal_value
+    inverted = color.invert(True)
+    nt.assert_array_equal(
+        inverted._internal_value, (1 - rgba[0], 1 - rgba[1], 1 - rgba[2], 1 - rgba[3])
+    )
+
+
+def test_interpolate() -> None:
+    r1 = RED._internal_value
+    r2 = YELLOW._internal_value
+    nt.assert_array_equal(
+        RED.interpolate(YELLOW, 0.5)._internal_value, 0.5 * r1 + 0.5 * r2
+    )
+
+
+def test_opacity() -> None:
+    nt.assert_equal(RED.opacity(0.5)._internal_value[3], 0.5)
+
+
+def test_parse() -> None:
+    nt.assert_equal(ManimColor.parse([RED, YELLOW]), [RED, YELLOW])
+
+
+def test_mc_from_functions() -> None:
+    color = ManimColor.from_hex("#ff00a0")
+    nt.assert_equal(color.to_hex(), "#FF00A0")
+
+    color = ManimColor.from_rgb((1.0, 1.0, 0.0))
+    nt.assert_equal(color.to_hex(), "#FFFF00")
+
+    color = ManimColor.from_rgba((1.0, 1.0, 0.0, 1.0))
+    nt.assert_equal(color.to_hex(True), "#FFFF00FF")
+
+    color = ManimColor.from_hsv((1.0, 1.0, 1.0), alpha=0.0)
+    nt.assert_equal(color.to_hex(True), "#FF000000")
+
+
+def test_hsv_init() -> None:
+    color = HSV((0.25, 1, 1))
+    nt.assert_array_equal(color._internal_value, np.array([0.5, 1.0, 0.0, 1.0]))
+
+
+def test_into_HSV() -> None:
+    nt.assert_equal(RED.into(HSV).into(ManimColor), RED)
```

Comment:
Is adding colors/performing other operations on colors supported? Those might also be good tests to make sure it works for both `ManimColor` and other color spaces (`HSV`).

Reply:
yip, just haven't written the tests by now and still unsure of how useful it would be, but it works!